File: curso/modulo-4/aula-04/imprevisivel.py
import hashlib
from blake3 import blake3

mensagens = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']

for mensagem in mensagens:
    print('SHA-256:', mensagem, ' - ', hashlib.sha256(mensagem.encode('utf-8')).hexdigest())


for mensagem in mensagens:
    print('SHA-512:', mensagem, ' - ', hashlib.sha512(mensagem.encode('utf-8')).hexdigest())


for mensagem in mensagens:
    print('Blake3:', mensagem, ' - ', blake3(mensagem.encode('utf-8')).hexdigest())

File: curso/modulo-4/aula-04/imprevisivel.cs
#r "nuget: Blake3, 0.5.0"

using System.Security.Cryptography;
using System.Text;

var mensagens = new[] { "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z" };

using (var sha256Hash = SHA256.Create())
{
    foreach (var mensagem in mensagens)
    {
        var hash = sha256Hash.ComputeHash(Encoding.ASCII.GetBytes(mensagem));
        Console.WriteLine("SHA-256: {0}", BitConverter.ToString(hash).ToLower().Replace("-", string.Empty));
    }
}

using (var sha512Hash = SHA512.Create())
{
    foreach (var mensagem in mensagens)
    {
        var hash = sha512Hash.ComputeHash(Encoding.ASCII.GetBytes(mensagem));
        Console.WriteLine("SHA-512: {0}", BitConverter.ToString(hash).ToLower().Replace("-", string.Empty));
    }
}

foreach (var mensagem in mensagens)
{
    var hash = Blake3.Hasher.Hash(Encoding.UTF8.GetBytes(mensagem));
    Console.WriteLine("Blake3: {0}", hash);
}